nodebuilder/node/node.go

```diff
@@ -0,0 +1,64 @@
+package node
+
+import (
+	"context"
+
+	"github.com/filecoin-project/go-jsonrpc/auth"
+)
+
+var (
+	buildTime       string
+	lastCommit      string
+	semanticVersion string
+)
+
+// Module defines the API related to interacting with the "administrative"
+// node.
+//
+//go:generate mockgen -destination=mocks/api.go -package=mocks . Module
+type Module interface {
+	// Type returns the node type.
+	Type(context.Context) Type
```

Comment:
Do we need to implement JSONMarshal and Unmarshal for the type? It seems like it will work just by marshaling the enum number, but we may want to have human-readable